File: Code Generation/responses/cpp/claude/medium/choose-k-elements-with-maximum-sum.cpp
#include <vector>
#include <algorithm>
#include <queue>
using namespace std;

class Solution {
public:
    vector<long long> findMaxSum(vector<int> nums1, vector<int> nums2, int k) {
        int n = nums1.size();
        vector<int> idx(n);
        iota(idx.begin(), idx.end(), 0);
        sort(idx.begin(), idx.end(), [&](int a, int b){
            return nums1[a] < nums1[b];
        });
        
        vector<long long> ans(n, 0);
        priority_queue<int, vector<int>, greater<int>> minHeap; // min-heap of size k
        long long heapSum = 0;
        
        int i = 0;
        while (i < n) {
            // Group all indices with same nums1 value
            int j = i;
            while (j < n && nums1[idx[j]] == nums1[idx[i]]) j++;
            
            // For all in [i, j), the heap currently holds eligible nums2 values
            for (int t = i; t < j; t++) {
                ans[idx[t]] = heapSum;
            }
            
            // Now add nums2 values for indices in [i, j) to heap
            for (int t = i; t < j; t++) {
                int val = nums2[idx[t]];
                minHeap.push(val);
                heapSum += val;
                if ((int)minHeap.size() > k) {
                    heapSum -= minHeap.top();
                    minHeap.pop();
                }
            }
            
            i = j;
        }
        
        return ans;
    }
};

File: Code Generation/responses/python/claude/medium/choose-k-elements-with-maximum-sum.py
from typing import List
import heapq

class Solution:
    def findMaxSum(self, nums1: List[int], nums2: List[int], k: int) -> List[int]:
        n = len(nums1)
        # Sort indices by nums1 value
        order = sorted(range(n), key=lambda i: nums1[i])
        
        answer = [0] * n
        min_heap = []  # min-heap of nums2 values seen so far (with smaller nums1)
        heap_sum = 0
        
        i = 0
        while i < n:
            # Group indices with the same nums1 value
            j = i
            while j < n and nums1[order[j]] == nums1[order[i]]:
                j += 1
            
            # For all indices in this group, compute answer using current heap
            for idx in range(i, j):
                answer[order[idx]] = heap_sum
            
            # Now add nums2 values for this group into the heap
            for idx in range(i, j):
                val = nums2[order[idx]]
                heapq.heappush(min_heap, val)
                heap_sum += val
                if len(min_heap) > k:
                    heap_sum -= heapq.heappop(min_heap)
            
            i = j
        
        return answer